packages/hardhat-core/src/internal/cli/autocomplete.ts

```diff
@@ -62,50 +95,226 @@ export async function complete({
   }
 
   if (prev === "--network") {
-    return Object.keys(hre.config.networks);
+    return networks.filter(startsWithLast);
   }
 
   // if the previous word is a param, then a value is expected
   // we don't complete anything here
   if (prev.startsWith("-")) {
     const paramName = ArgumentsParser.cLAToParamName(prev);
 
-    const globalParam: any = (HARDHAT_PARAM_DEFINITIONS as any)[paramName];
+    const globalParam = HARDHAT_PARAM_DEFINITIONS[paramName as GlobalParam];
     if (globalParam !== undefined && !globalParam.isFlag) {
-      return [];
+      return filesystemSuggestions(last);
     }
   }
 
   // if there's no task, we complete either tasks or params
-  if (task === undefined || hre.tasks[task] === undefined) {
-    const tasks = Object.values(hre.tasks)
+  if (task === undefined || tasks[task] === undefined) {
+    const taskNames = Object.values(tasks)
       .map((x) => x.name)
       .filter((x) => !x.includes(":"));
```

Comment:
maybe store if the task is internal or not and filter based on that?

Reply:
Not sure I understood this. You mean changing this in core to have that information here?

Reply:
I think the `includes(":")` is used to decide if a task name should be suggested or not, as a proxy to whether a task is internal or not. That info, being internal or not, is already present in the task definition object.

Reply:
For example, the help printer does something very similar.

Reply:
Got it. Fixed in a508a9c76801ea391cc6ccb1012993ab23fe3698